### ISSUE TITLE
Move on to Rust 2018

### ISSUE THREAD

**Issue by drozdziak1** [sigp/lighthouse#122]
## Description
Most of Rust 2018 edition has recently made it to stable https://blog.rust-lang.org/2018/12/06/Rust-1.31-and-rust-2018.html and it should be easy enough to move `lighthouse` on to it. The switch is **backwards-incompatible, but at dependency level everything is 100% compatible**, i.e. if you have older edition deps they'll work and vice versa (provided your `rustc` knows how to use Rust 2018)

## Present Behaviour
Everything is still in 2015 (i.e. there's no `edition = 2018`

## Expected Behaviour
It should be nice to keep up as among other things, Rust 2018 makes trait objects use more explicit

## Steps to resolve
Here's a page from the Edition Guide that walks through the process of moving on to 2018
https://rust-lang-nursery.github.io/edition-guide/editions/transitioning-an-existing-project-to-a-new-edition.html

**Comment by paulhauner** [comment]
I completely agree. @ralexstokes and I were discussing this in the [gitter channel](http://gitter.im/sigp/lighthouse) recently.

I think we casually decided that we'll start new crates with `edition = "2018"` and then do a mass upgrade at some point.

I'm more than happy for you to be the mass-upgrader if you'd like @drozdziak1  :)

**Comment by drozdziak1** [comment]
FYI started on this
